tests/unit/test_config_provider.py

```diff
@@ -363,6 +363,20 @@ def test_can_get_config_provider_non_chain_provider(self):
         self.assertIsInstance(provider, ConstantProvider)
         self.assertEqual(value, 'bar')
 
+    def test_deepcopy_preserves_overrides(self):
+        provider = ConstantProvider(100)
+        config_store = ConfigValueStore(mapping={'fake_variable': provider})
+        config_store.set_config_variable('fake_variable', 'foo')
+
+        config_store_deepcopy = copy.deepcopy(config_store)
+
+        self.assertDictEqual(
+            {'fake_variable': 'foo'}, config_store_deepcopy._overrides
+        )
```

Comment:
Let's avoid using internal members in the tests as they are not public interfaces. I'd say we should just remove this line and rely on the `get_config_variable` interface.